### PR TITLE
docs: Proposed CHANGELOG with one-line entries

### PR DESCRIPTION
**Pull request by richb-hanover**
See https://github.com/PRQL/prql/issues/2595 for details

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,13 @@
...
 ## 0.8.2 — [unreleased]
 
+_The following unreleased version are only available in the `main` branch. They
+will become the public version at the next release._
+
 **Features**:
 
-- Add a `~=` regex search operator (@max-sixty, #2458). An example:
+- Add a `~=` regex search operator (@max-sixty,
+  https://github.com/PRQL/prql/pull/2458). An example:
```
Comment (max-sixty) [inline CHANGELOG.md]: GitHub will make these automatically. Though they don't show in https://prql-lang.org/book/changelog.html admittedly.

Obv fine to have them here for now, but to keep consistent with the remainder of the doc, I'll probably switch to the existing convention before publishing.

In the short term, we could have a sentence at the top pointing folks towards https://github.com/PRQL/prql/releases, where these are formatted.

In the medium term, we could add this as a preprocessor (or maybe one exists out there?) to make these into links.

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,6 +28,15 @@
 
   ...though the exact form differs by dialect.
 
+- Change to function syntax: `let f = x -> ...` See
+  https://github.com/PRQL/prql/blob/main/web/book/src/queries/functions.md
```
Comment (max-sixty) [inline CHANGELOG.md:31]: Why point to GitHub rather than the Book directly?

**Review by max-sixty — COMMENTED** [review]
Cheers @richb-hanover

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,9 +2,13 @@
 
 ## 0.8.2 — [unreleased]
 
+_The following unreleased version are only available in the `main` branch. They
```
Comment (max-sixty) [inline CHANGELOG.md]: This had seemed implicit from "Unreleased", but maybe that's becaues I'm used to it...

File: CHANGELOG.md (hunk 1 above)
Comment (richb-hanover) [reply CHANGELOG.md]: You're right - I switched it back.

File: CHANGELOG.md (hunk 2 above)
Comment (richb-hanover) [reply CHANGELOG.md:31]: As interim documentation for this unreleased change, I preferred to point to the page where I was certain the information exists.

File: CHANGELOG.md (hunk 3 above)
Comment (richb-hanover) [reply CHANGELOG.md]: I added this sentence to make it clear where someone _could_ find these features (in the `main` branch), even though they're not in any stable branch.

And I'm treating this PR as an experiment: I wanted to see how little information was required of a developer to leave hints about the unreleased stuff. 

I'm willing to be fairly forgiving in the format of the Unreleased section: a one-liner that points to a PR, an issue, or even a discussion (github or discord) that talks about what has actually been implemented is useful. Thanks.

File: CHANGELOG.md [hunk 4]
```diff
@@ -24,6 +27,15 @@
 
   ...though the exact form differs by dialect.
 
+- Change to function syntax: `let f = x -> ...` See
+  https://github.com/PRQL/prql/blob/main/web/book/src/queries/functions.md
```
Comment (eitsupi) [inline CHANGELOG.md:31]:
```suggestion
  <https://github.com/PRQL/prql/blob/main/web/book/src/queries/functions.md>
```

Note that links generally need to be enclosed in `<>`.

Comment (richb-hanover) [reply CHANGELOG.md:31]: I don't see a difference with or without the `< ... >`. For the Unreleased section, I'm content to be flexible about what people post there - it'll be updated/re-written as we approach a release.

Comment (eitsupi) [reply CHANGELOG.md:31]: Please check <https://spec.commonmark.org/0.30/#links> and rust-lang/mdBook#1114

Comment (max-sixty) [reply CHANGELOG.md:31]: - It's OK to just have one liners here as reminders for someone else to do it
- Ultimately these should be formatted links to the book like [Functions docs](https://prql-lang.org/book/queries/functions.html)
- To the extent we can do that way initially, that's ideal!

File: CHANGELOG.md (hunk 2 above)
Comment (max-sixty) [reply CHANGELOG.md:31]: OK — re https://github.com/PRQL/prql/pull/2613/files#r1204663635 — helpful but would be even more helpful in a final state!

**Comment by richb-hanover** [comment]
Thanks!

**Comment by max-sixty** [comment]
> Thanks!

Thank you!

**Pull request merged**